lib/portage/binpkg.py

```diff
@@ -2,19 +2,32 @@
 # Distributed under the terms of the GNU General Public License v2
 
 import tarfile
+from portage import os
 from portage.const import SUPPORTED_XPAK_EXTENSIONS, SUPPORTED_GPKG_EXTENSIONS
+from portage.exception import InvalidBinaryPackageFormat
 from portage.output import colorize
 from portage.util import writemsg
 
 
-def get_binpkg_format(binpkg_path):
+def get_binpkg_format(binpkg_path, remote=False):
     if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
         file_ext_format = "xpak"
     elif binpkg_path.endswith(SUPPORTED_GPKG_EXTENSIONS):
         file_ext_format = "gpkg"
     else:
         file_ext_format = None
 
+    if remote:
+        if file_ext_format is not None:
+            return file_ext_format
+        else:
+            raise InvalidBinaryPackageFormat(
+                f"Unsupported binary package format from '{binpkg_path}'"
+            )
+
+    if file_ext_format is not None:
+        return file_ext_format
```

Comment:
Now I'm confused. When called on a local binpkg, the function would previously check both suffix and content, and report a mismatch. Now it will always use the suffix, so the content check would only be used if someone misnamed the package (would that even work?), and the part about suffix/format mismatch would never be used.

Reply:
This can lead to a lot of extra IO. especially if you have many versions, but only one of them will be used in the end.
I will add a read check only if the format is invalid.